tests/datasets/causal_data/test_causal_utility_functions.py

```
@@ -0,0 +1,166 @@
+
+    # Test specific dataset type directory
+    diamond_dir = get_causal_data_dir('diamond')
+    assert diamond_dir.endswith('.fintorch_data/causal/diamond')
+    assert os.path.isabs(diamond_dir)
+
+
+def test_clear_causal_data():
+    """Test the clear_causal_data function."""
+    with tempfile.TemporaryDirectory() as temp_dir:
+        # Patch the default directory to use temporary directory
+        with patch('os.path.expanduser', return_value=temp_dir):
```

Ruff (0.11.9)

Use a single `with` statement with multiple contexts instead of nested `with` statements

(SIM117)